### PR TITLE
mod: bump to lnd v0.18.2

### PR DESCRIPTION
**Pull request by bitromortac**
Not sure if this is too large of a jump from 0.16 to 0.18.

The change in `hashmailserver_harness.go` is needed in order to be able to run tests in a dependent repository.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -148,7 +148,7 @@ itest: itest-run
 
 itest-run:
 	@$(call print, "Running integration tests.")
-	$(GOTEST) ./itest -tags="$(ITEST_TAGS)" $(TEST_FLAGS) -logoutput -logdir=itest_logs
```
Comment (bitromortac) [inline Makefile:151]: It seems like this didn't generate any logs previously and the flags are not present anymore.

**Review by ellemouton — COMMENTED** [review]
Nice 🙏  can we open a lit PR referencing this so we can see it passing in itests on that side?

**Review by jamaljsr — APPROVED** [review]
tACK LGTM 👌

Tested connecting to a regtest `litd` node and making a bunch of RPC calls. I didn't run into any issues.

**Comment by bitromortac** [comment]
> Nice 🙏 can we open a lit PR referencing this so we can see it passing in itests on that side?

I opened the PR: https://github.com/lightninglabs/lightning-terminal/pull/820, tests are passing :rocket:

**Review by ellemouton — APPROVED** [review]

**Review by ViktorTigerstrom — APPROVED** [review]
tACK LGTM 🚀

**Pull request merged**